gcloud/_helpers.py

```diff
@@ -159,6 +161,41 @@ def _app_engine_id():
     return app_identity.get_application_id()
 
 
+def _file_project_id():
+    """Gets the project id from the credentials file if one is available.
+
+    :rtype: str or ``NoneType``
+    :returns: Project-ID from JSON credentials file if value exists,
+              else ``None``.
+    """
+    credentials_file_path = os.getenv('GOOGLE_APPLICATION_CREDENTIALS')
+    if credentials_file_path:
+        with open(credentials_file_path, 'r') as credentials_file:
+            credentials_data = credentials_file.read()
+            credentials = json.loads(credentials_data)
+            return credentials.get('project_id')
+    else:
+        return None
+
+
+def _default_service_project_id():
+    """Retrieves the project ID from the gcloud command line tool.
+
+    :rtype: str or ``NoneType``
+    :returns: Project-ID from ``gcloud info`` else ``None``
+    """
+    command = subprocess.Popen(['gcloud info'],
```

Comment:
If you run `gcloud info` you can see the `Active Configuration Path: [~/.config/gcloud/configurations/config_default]` so I thought it could change. But if we can rely on the `~/.config/gcloud/configurations/config_default` path working then I can just use ConfigParser to get the `project` property from it.